src/main/java/build/buildfarm/instance/server/NodeInstance.java

```diff
@@ -655,7 +655,7 @@ private static ListenableFuture<Digest> downloadUrl(
       // not great, plenty risky for large objects
       ByteString data;
       try (InputStream in = connection.getInputStream()) {
-        data = ByteString.readFrom(connection.getInputStream());
+        data = ByteString.readFrom(in);
```

Comment:
This is actually a major bug, can you put this in separately as a PR? - it leaks the second stream.

Reply:
https://github.com/bazelbuild/bazel-buildfarm/pull/1784